docs/src/components/ComponentDoc/ComponentExample/ComponentExample.js

```diff
@@ -64,36 +62,44 @@ class ComponentExample extends PureComponent {
     renderHtml: true,
   }
 
-  componentWillMount() {
-    const { examplePath } = this.props
-    this.anchorName = examplePathToHash(examplePath)
-
-    this.setState({
-      showCode: this.isActiveHash(),
-      sourceCode: this.getOriginalSourceCode(),
-    })
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { examplePath, exampleSources, location } = nextProps
-    const nextSourceCode = exampleSources[examplePath]
+  static getDerivedStateFromProps(nextProps, prevState = {}) {
+    const {
+      exampleKeys,
+      examplePath,
+      exampleSources,
+      location: { hash },
+    } = nextProps
+
+    const anchorName = examplePathToHash(examplePath)
+    const isActiveHash = anchorName === getFormattedHash(exampleKeys, hash)
```

Comment:
As we don't support old hashes more, we can safely remove `getFormattedHash()` fully:

```suggestion
    const isActiveHash = anchorName === location.hash
```

Reply:
It also seems that it's a single place where we are using `exampleKeys`, so we can remove them, too 💯

Reply:
Done